events-api/src/modules/events/events.repository.ts

```diff
@@ -18,4 +21,156 @@ export class EventsRepository {
 
     return result;
   }
+
+  static async getOne(eventId: string): Promise<EventModel[]> {
+    const result = db.execute<EventModel>(
+      `
+        SELECT
+            event_id as eventId,
+            event_name as eventName,
+            event_description as eventDescription,
+            event_location as eventLocation,
+            event_date as eventDate,
+            event_created_at as eventCreatedAt,
+            event_updated_at as eventUpdatedAt
+        FROM
+            events
+        WHERE event_id = ?`,
+      [eventId]
+    );
+
+    return result;
+  }
+
+  static async create(eventData: NewEvent): Promise<EventModel[]> {
+    const id = uuidv4();
+
+    const result = db.execute<EventModel>(
+      `
+        INSERT INTO events (
+          event_id, event_name, event_description, event_location, event_date
+        ) VALUES (?, ?, ?, ?, ?)
+        RETURNING
+          event_id as eventId,
+          event_name as eventName,
+          event_description as eventDescription,
+          event_location as eventLocation,
+          event_date as eventDate,
+          event_created_at as eventCreatedAt,
+          event_updated_at as eventUpdatedAt`,
+      [
+        id,
+        eventData.eventName,
+        eventData.eventDescription,
+        eventData.eventLocation,
+        eventData.eventDate,
+      ]
+    );
+
+    return result;
+  }
+
+  static async update(
+    eventId: string,
+    newData: Partial<NewEvent>
+  ): Promise<EventModel> {
+    const oldData = await db.execute<EventModel>(
+      `
+        SELECT
+          event_id as eventId,
+          event_name as eventName,
+          event_description as eventDescription,
+          event_location as eventLocation,
+          event_date as eventDate
+        FROM
+          events
+        WHERE
+          event_id = ?`,
+      [eventId]
+    );
+
+    if (oldData.length === 0) {
+      throw new Error("Event not found");
+    }
+
+    const updatedData = {
+      ...oldData[0],
+      ...newData,
+      eventUpdatedAt: formatDateTime(new Date()),
+    };
+
+    await db.execute(
+      `
+      UPDATE events SET
+        event_name = ?,
+        event_description = ?,
+        event_location = ?,
+        event_date = ?,
+        event_updated_at = ?
+      WHERE event_id = ?
+    `,
+      [
+        updatedData.eventName,
+        updatedData.eventDescription,
+        updatedData.eventLocation,
+        updatedData.eventDate,
+        updatedData.eventUpdatedAt,
+        eventId,
+      ]
+    );
+
+    return updatedData as EventModel;
```

Comment:
Don't use type casting

Reply:
Also better to return actual updated event instead of trying to "mimic" the updated data